File: core/commands/registry/irules/udp__local_port.py
# Enriched from F5 iRules reference documentation.
"""UDP::local_port -- Returns the local UDP port/service number."""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef, make_av
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/UDP__local_port.html"


_av = make_av(_SOURCE)


@register
class UdpLocalPortCommand(CommandDef):
    name = "UDP::local_port"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="UDP::local_port",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Returns the local UDP port/service number.",
                synopsis=("UDP::local_port (clientside | serverside)?",),
                snippet="Returns the local UDP port/service number.",
                source=_SOURCE,
                examples=(
                    "when CLIENT_ACCEPTED {\n"
                    "  if { [matchclass [UDP::local_port] equals $::ValidUDPPorts ] } {\n"
                    "    pool udp_pool\n"
                    "  } else {\n"
                    "     discard\n"
                    "  }\n"
                    "}"
                ),
                return_value="Returns the local UDP port/service number",
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="UDP::local_port (clientside | serverside)?",
                    arg_values={
                        0: (
                            _av(
                                "clientside",
                                "UDP::local_port clientside",
                                "UDP::local_port (clientside | serverside)?",
                            ),
                            _av(
                                "serverside",
                                "UDP::local_port serverside",
                                "UDP::local_port (clientside | serverside)?",
                            ),
                        )
                    },
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(transport="udp"),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.UDP_STATE,
                    reads=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/udp__local_port.rs
// tcl-lsp — a language server and toolchain for Tcl
// Copyright (C) 2026 James Deucker (bitwisecook) <https://github.com/bitwisecook>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
// SPDX-License-Identifier: AGPL-3.0-or-later

//! `UDP::local_port` iRules command.
use crate::prelude::*;
use tcl_dialect::model::SpecSurface;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "UDP::local_port",
        surface: Some(SpecSurface::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "Returns the local UDP port/service number.",
            synopsis: &["UDP::local_port (clientside | serverside)?"],
            snippet: "Returns the local UDP port/service number.",
            source: "https://clouddocs.f5.com/api/irules/UDP__local_port.html",
            examples: "when CLIENT_ACCEPTED {\n  if { [matchclass [UDP::local_port] equals $::ValidUDPPorts ] } {\n    pool udp_pool\n  } else {\n     discard\n  }\n}",
            return_value: "Returns the local UDP port/service number",
        }),
        event_requires: Some(EventRequires {
            client_side: false,
            server_side: false,
            transport: Some("udp"),
            profiles: &[],
            also_in: &[
                "SIP_REQUEST",
                "SIP_REQUEST_SEND",
                "SIP_RESPONSE",
                "STREAM_MATCHED",
            ],
            flow: false,
        }),
        forms: &[FormSpec {
            synopsis: "UDP::local_port (clientside | serverside)?",
            ..FormSpec::DEFAULT
        }],
        side_effects: &[SideEffect {
            target: SideEffectTarget::UdpState,
            reads: true,
            connection_side: ConnectionSide::Both,
            ..SideEffect::DEFAULT
        }],
        ..CommandSpec::DEFAULT
    }
}
